testing/tests/DevExpress.ui.widgets.editors/dateRangeBox.tests.js

```diff
@@ -180,4 +191,209 @@ QUnit.module('DateRangeBox Initialization', moduleConfig, () => {
             });
         });
     });
+
+    QUnit.module('DateBox\'s option dependency from DateRangeBox options ', () => {
+        const initialDateRangeBoxOptions = {
+            showClearButton: true,
+            showDropDownButton: true,
+            buttons: ['dropDown'],
+            // TODO: extend this list of options
+        };
+
+        QUnit.test('StartDateBox has expected settings', function(assert) {
+            this.reinit(initialDateRangeBoxOptions);
+
+            const expectedOptions = {
+                showClearButton: true,
+                showDropDownButton: false,
+                buttons: undefined,
+            };
+            const startDateBox = getStartDateBoxInstance(this.instance);
+
+            Object.entries(initialDateRangeBoxOptions).forEach(([key]) => {
+                assert.deepEqual(expectedOptions[key], startDateBox.option(key), `${key} value is correct`);
+            });
+        });
+
+        QUnit.test('EndDateBox has expected settings', function(assert) {
+            this.reinit(initialDateRangeBoxOptions);
+
+            const expectedOptions = {
+                showClearButton: true,
+                showDropDownButton: false,
+                buttons: undefined,
+            };
+            const endDateBox = getEndDateBoxInstance(this.instance);
+
+            Object.entries(initialDateRangeBoxOptions).forEach(([key]) => {
+                assert.deepEqual(expectedOptions[key], endDateBox.option(key), `${key} value is correct`);
+            });
+        });
+    });
+});
+
+QUnit.module('DropDownButton', moduleConfig, () => {
+    QUnit.test('Drop down button should be rendered if showDropDownButton is true', function(assert) {
+        this.reinit({
+            showDropDownButton: true
+        });
+
+        assert.strictEqual(getButtons(this.$element).length, 1, 'drop down button was rendered');
+    });
+
+    QUnit.test('Drop down button should not be rendered if showDropDownButton is false', function(assert) {
+        this.reinit({
+            showDropDownButton: false
+        });
+
+        assert.strictEqual(getButtons(this.$element).length, 0, 'drop down button was not rendered');
+    });
+
+    QUnit.test('Drop button should not be rendered if buttons is [] and showDropDownButton is true', function(assert) {
```

Comment:
update in all tests, plz
```suggestion
    QUnit.test('Drop down button should not be rendered if buttons is [] and showDropDownButton is true', function(assert) {
```

Reply:
https://github.com/DevExpress/DevExtreme/pull/24306/commits/0ece794aafd40022a99d12436a248615e63456f5